### PR TITLE
feature(logout):add user logout router

### PR DESCRIPTION
**Pull request by SteveFunso**
#### What does this PR do?
Enables users to log out of the application

#### Description of Task to be completed?
When a user clicks on the logout link, they are signed out of the application with their session ended

#### What are the relevant pivotal tracker stories?
#167727312

### CODE REVIEW

File: src/routes/logout.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// GET /logout
+router.get('/logout', function(req, res, next) {
+    if (req.session) {
+      // delete session object
+      req.session.destroy(function(err) {
+        if(err) {
+          return next(err);
+        } else {
+          return res.redirect('/');
+        }
```
[the remark this answers is not in the document]
Comment (Hector101) [reply src/routes/logout.js]: @SteveFunso Kindly fix all Eslint warnings, hence fixing the hound error flags on this PR.

**Review by Hector101 — CHANGES_REQUESTED** [review]

File: src/routes/logout.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (SteveFunso) [reply src/routes/logout.js]: Ok... noted

File: src/routes/logout.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import express from 'express';
+
+const router = express.Router();
+
+// GET /logout
+router.get('/logout', function(req, res, next) {
+    if (req.session) {
+      // delete session object
+      req.session.destroy(function(err) {
```
Comment (Hector101) [inline src/routes/logout.js]: I thought the agreement was to use JWT over sessions?

**Review by Hector101 — CHANGES_REQUESTED** [review]

**Review by joaquinto — COMMENTED** [review]
Nice work @SteveFunso however you need to work with `jwt` and not `session` since we are working on a `stateless API`. All you need to do is send some kind of response to the front end. It is from the frontend we are going to clear the `jwt token`. One other thing is you didn't resolve the conflicts well. Please look into that. Nice work bro.

File: src/helpers/auth.js~merged [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Verify a token by using a secret key and a public key.
+* @param {Object} token
+* @return {Object} return verified token
+*/
+  static verifyToken(token) {
+    return jwt.verify(token, SECRET_KEY);
+  }
+}
+>>>>>>> e9c3165... feat(user-login-endpoint):user should be able to login with a valid email and password
+export default Auth;
```
Comment (joaquinto) [inline src/helpers/auth.js~merged]: Please attend to this. Thank you.

**Review by steelze — DISMISSED** [review]

**Review by joaquinto — DISMISSED** [review]
Welcome bro. Thumbs up

**Review by shonubijerry — CHANGES_REQUESTED** [review]
@joaquinto made a comment on the auth.js-merged file, Also get rid of the hound errors. I think most of them are your new line format. Check bottom right of your VSCode and change LF to CELF.

File: src/modules/auth/auth.routes.js [hunk 4]
```diff
@@ -5,4 +5,6 @@ import ValidateAuth from '../../middlewares/ValidateAuth';
 const router = express.Router();
 router.post('/signin', ValidateAuth.validateSigninFormData, AuthController.signin);
 
+router.get('/logout', (req, res) => res.status(200).json({ status: '204', message: 'Successfully Logged out' }));
```
Comment (Hector101) [inline src/modules/auth/auth.routes.js]: @SteveFunso kindly move ```(req, res) => res.status(200).json({ status: '204', message: 'Successfully Logged out' })``` to a controller file and import to the route.

Also, ensure to make use of the `responseHandler` helper methods for consistency with our response

**Review by Hector101 — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -4,14 +4,14 @@
   "description": "Making company travel and accomodation easy and convinient",
   "main": "index.js",
   "scripts": {
-    "test": "npm run migration:refresh:test && NODE_ENV=test npm run db:seed && NODE_ENV=test nyc --reporter=html --reporter=text mocha --timeout 20000 -r @babel/register ./src/modules/**/*.test.js --exit",
+    "test": "npm run migration:refresh:test && cross-env NODE_ENV=test npm run db:seed && cross-env NODE_ENV=test nyc --reporter=html --reporter=text mocha --timeout 20000 -r @babel/register ./src/modules/**/*.test.js --exit cross-env NODE_ENV=test",
```
Comment (shonubijerry) [inline package.json]: @SteveFunso Kindly remove the `cross-env NODE_ENV=test ` after `--exit`

**Review by shonubijerry — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (SteveFunso) [reply package.json]: ok...  noted

File: src/modules/auth/auth.routes.js (hunk 4 above)
Comment (SteveFunso) [reply src/modules/auth/auth.routes.js]: ok noted.

**Review by shonubijerry — DISMISSED** [review]

File: src/middlewares/validateAuth.js [hunk 6]
```diff
@@ -3,26 +3,31 @@ import joiValidator from '../helpers/joiValidator';
...
    * validate email and password
    * @param {Object} req
    * @param {Object} res
    * @param {Function} next
    * @returns {Object} error
    */
-const validateSigninFormData = (req, res, next) => {
-  const password = Joi.string().required();
-  const email = Joi.string().email().required();
+  static validateSigninFormData(req, res, next) {
```
Comment (Hector101) [inline src/middlewares/validateAuth.js]: @SteveFunso Kindly revert these changes. using classes is not recommended in this case.

**Review by Hector101 — CHANGES_REQUESTED** [review]
